packages/itwinui-react/src/core/Dialog/DialogMain.tsx

```diff
@@ -205,10 +205,11 @@ export const DialogMain = React.forwardRef((props, ref) => {
     <CSSTransition
       in={isOpen}
       classNames={{
-        enter: 'iui-dialog-animation-enter',
-        enterActive: 'iui-dialog-animation-enter-active',
+        enter: styles['iui-dialog-animation-enter'],
+        enterActive: styles['iui-dialog-animation-enter-active'],
+        enterDone: styles['iui-dialog-visible'],
       }}
-      timeout={{ exit: 600 }}
+      timeout={{ enter: 100, exit: 600 }}
```

Comment:
Did you investigate why they were not appearing? We shouldn't add a random delay like this. It's a hack, not a proper fix.

Reply:
Actually, timeout for `enter` is not required. But timeout for `exit` seems like its required.

`100` is too small a timeout for `exit`. i.e. the dialog animation is abrupt and the dialog never goes all the way to the bottom of the screen. Since `600` worked before, we could keep that. Additionally, you could also investigate in CSS how much time is really needed. If it's smaller than `600`, then we can reduce this number.